### ISSUE TITLE
[Feature Request] API tweaks

### ISSUE THREAD

**Issue by mli** [autogluon/autogluon#1969]
A few minor API tweaks, which I believe to improve usability. I'd like to add it to the 0.6 milestone, but not necessary to address all by the release. I'll probably add more in future

## Tabular

- [ ] `evaluate` accepts `auxiliary_metrics=True/False` while `leaderboard` has a `extra_metrics` to accept a list of metrics. it's good to unify them, e.g. both support these two arguments, or just keep one

- [x] `silent=False` in default for `evaluate` and `leaderboard`. i suggest to change it to true, because the two common cases are 1/ `res = predict.evaluate(...)` or run `predict.evaluate(...)` in jupyter notebook to print the results. Now we need to always set `silent=True` to avoid double printing. 

- [x] metrics are adopted from scikit-learn, but removed the `_score` and `_loss` postfix for most metrics for simplicity. there are still some of them such as `pac_score`, we could rename it to `pac`.  

- [x] when using `hyperparameter_tune_kwargs` in `fit`, we must schedule `scheduler` and `searcher`. we can set default values `local` and `auto`.

- [ ] `fit` accepts `**kwargs`,  it's better to explicitly write down what kwargs it accepts 

In general, our methods have a long list of arguments. Comparing to others such as scikit-learn or pytorch, our API doc is harder to read. Maybe we should simplify the API doc, or organize them into sections. 

## MultiModal  

One main issue I found is multimodal's API is different than tabular in several places, e.g. argument names and orders. They can be interchangeably used in several tasks, as they all accept dataframe as inputs. I imagine that users may just replace `TabularPredictor` to `MultiModalPredictor` during prototyping. (you can use the `multimodal` option in tabular, but using MultiModalPredictor alone often runs faster.) So making their API similar to each others, at least for basic usages, will benefit users.

- [ ] documents call dataset from other modules, such as `vision.ImageDataset`. We can create alias in the multimodal as well. 

  - [x] when you installed both pytorch and mxnet, gluoncv will warn you about the memory usage. i feel we should fix its issue instead of warning, e.g. only import pytorch even when mxnet exists

- [ ] `default_metric` is auc for binary classification, with a warning that it will be slow. while not use accuracy in default for all classification ? ( tabular uses accuracy.) 

- [ ] `evaluate` has a `metrics`, which is insistent with tabular.  

- [ ] continuously training in multimodal is calling `fit` twice or `load` then `fit`, while tabular use `fit_extra` and raise error when you call `fit` twice. we should make them more similarly behave. also multimodal overwrite model when no `save_path` specify.

- [ ]  `evaluate` and `predict` show the progress_bar in default, while it's ok for training, but annoying for inference. right now you can only set `enable_progress_bar` when creating. i suggest to disable it in default for inference.

  - [ ] also sometimes got error like
       > /pytorch_lightning/loops/utilities.py:91: PossibleUserWarning: `max_epochs` was not set. Setting it to 1000 epochs. To train without an epoch limit, set `max_epochs=-1`.
      > rank_zero_warn(
